### PR TITLE
New Subtree version of website

### PR DESCRIPTION
**Pull request by dhutchis**
This is not ready for pulling to master yet.  The included docs of the child repositories do not include the following tags in their markdown files

``` Markdown

---
layout: default
title: MyriaX
group: "docs"
weight: 1

---
```

Therefore the `_includes/docs_list` does not include the child pages, and they do not appear in the top navigation menu.  Dominik, can you make this work with Jekyll?  It would be a shame to force all the other repository owners to put extra tags in their markdown.  We should be able to tell Jekyll how to handle them based on their directory location.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,9 +6,23 @@ See the site in action at [myria.cs.washington.edu](http://myria.cs.washington.e
 
 ## How to make changes
 
+**Warning: do not edit the subfolders in docs/.../** <br/>
+`docs/index.md` is okay to edit.
```
Comment (domoritz) [inline README.md:10]: Actually, it's also okay to edit any files that are not in folders, right?

File: docs/MyriaX/myriaX.markdown [hunk 2]
```diff
@@ -0,0 +1,228 @@
+Myria
```
Comment (domoritz) [inline docs/MyriaX/myriaX.markdown]: rename file to index.md

**Comment by domoritz** [comment]
I fixed the build but can't get the subtree to work. See below

```
~/D/U/myria-website ❯❯❯ bash subtree-pull.sh                                                                                                                                                       subtree
thisSubfolder: docs/myriax/
gitUrl       : git@github.com:uwescience/myria.git
gitName      : myria
gitBranch    : add_myriax_doc
gitSubdir    : docs/
origBranch   : subtree
tempBranch   : subtree-myria-add_myriax_doc-temp
Note: checking out 'myria/add_myriax_doc'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at c9186fd... Use the correct myriad doc. Use markdown extension.
Merge made by the 'ours' strategy.
Updated branch 'subtree-myria-add_myriax_doc-temp'
d6212d7dcf1b819425adefcb812d93a9e2be1517
Warning: you are leaving 1 commit behind, not connected to
any of your branches:

  82b2b37 Split 'docs/' into commit 'd6212d7dcf1b819425adefcb812d93a9e2be1517'

If you want to keep it by creating a new branch, this may be a good time
to do so with:

 git branch <new-branch-name> 82b2b37

Switched to branch 'subtree'
Your branch is up-to-date with 'origin/subtree'.
fatal: Unable to create '/Users/domoritz/Developer/UW/myria-website/.git/index.lock': File exists.

If no other git process is currently running, this probably means a
git process crashed in this repository earlier. Make sure no other git
process is running and remove the file manually to continue.
```

**Comment by domoritz** [comment]
Oh, and `/Users/domoritz/Developer/UW/myria-website/.git/index.lock` does not exits.

**Comment by dhutchis** [comment]
The `subtree-pull.sh` works fine for me.  Could you try inserting `sleep 1` after `git checkout "${origBranch}"` and seeing if that changes anything?  Maybe your file system isn't keeping up with the shell.

Yes it is okay to edit files in `docs/` but not subfolders.

**Comment by domoritz** [comment]
Doesn't fix it.

**Comment by dhutchis** [comment]
Not sure what to say. I wonder if this is dependent on Git versions. Just
to double check, you have no uncommitted changes, ya? If it doesn't work
for you, it probably will not work for others.

**Comment by domoritz** [comment]
> Not sure what to say. I wonder if this is dependent on Git versions. Just
> to double check, you have no uncommitted changes, ya? If it doesn't work
> for you, it probably will not work for others.

No uncommitted changes. Yeah, that's what I worry about. I am now against distributing the docs unless we get a stable approach working.

**Comment by dhutchis** [comment]
So.... let's review options.  The current status is that we can't figure out why the subtree pull script is not working for you.

I favor option 3.  What do you think?
1. Add post-commit-hooks in child repos that push docs to a dedicated subfolder of myria-website.
2. Build the myria-website manually and upload the built files to gh-pages, bypassing Guthub page's auto-building. Downside is that it requires building manually.  We can then use symlinks and github submodules.
3. Use git submodules in myria-website.  Add script in myria-website (called `update-docs-from-submodules.sh`) that expands each submodule and copies the docs/ data from each submodule into the appropriate path location for that child project.  Downside is that people need to know to execute this script in order to do the updates, as opposed to editing the files directly, since changes to the path we copy into will be overwritten (fortunately, this is captured in git history).
4. ...

**Comment by domoritz** [comment]
I think I know what broke things. I had gitx running. I'll try tomorrow
morning.

**Comment by domoritz** [comment]
The problem was gitx running. Works now.

**Comment by domoritz** [comment]
We discussed today that we want to go with this approach as it is giving us exactly the right directory structure.

**Comment by domoritz** [comment]
It would be awesome if we could also have subtree push. @dhutchis Do you have an idea how this could work? Then someone could edit the docs here and push the changes to the right repos (for example myria). That would make moving docs between repos a breeze.

**Comment by domoritz** [comment]
Can't merge right now. Can you resolve the conflicts and merge then?

**Comment by dhutchis** [comment]
Okay, this is good to go.  The merge was complicated because Jingjing edited the old, deleted docs/ submodule from myria-docs.  

Also fixed a trailing slash error that made git unhappy.

Dom, can you look one more time and merge if okay?

**Comment by domoritz** [comment]
Awesome. Can you announce this change in the meeting today and see that we assign tasks? I can't make it today, unfortunately.

**Pull request merged**